scripts/setup-ubuntu.sh

```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+sudo apt update
+sudo apt install -y --no-install-recommends \
+    git \
+    git-lfs \
+    python2.7 \
```

Comment:
I believe it should not need python2 anymore - looks like the README wasn't updated when we upgraded, sorry...

We use this action to build and deploy in the github workflow https://github.com/Exabyte-io/action-mkdocs-s3-publish and it seems to be using the default python for ubuntu-20.04 per https://github.com/Exabyte-io/action-mkdocs-s3-publish/blob/main/action.sh.

Reply:
Thank you @timurbazhirov. I could setup the project on ubuntu-20.04 (with default Python `3.8.10`). However, installation of python dependencies fails in ubuntu-22.04 (Python `3.10.6`). I can work on a ubuntu 20.04 container and figure out why it breaks in 22.04 later.

Reply:
Yep, this is exactly the behavior we seen before - this is because of the system python version difference between 20.04 and 22.04. We could avoid the dependency on the system python versions through `pyenv` (https://github.com/pyenv/pyenv). Internally, we use the following configuration to install it (Saltstack):

```yaml
pyenv-system-git:
  git.detached:
    - name: https://github.com/pyenv/pyenv.git
    - target: {{ _install_dir }}
    - rev: {{ pillar['python']['pyenv_version'] }}

pyenv_install_{{ python_version }}:
  cmd.run:
    - name: '{{ pyenv_install_path }}/bin/pyenv install --skip-existing {{ python_version }}'
    - env:
        - PYENV_ROOT: '{{ pyenv_install_path }}'
    - prepend_path: "{{ pyenv_install_path }}/bin:{{ pyenv_install_path }}/shims"
```

Then use python via pyenv during the build.

What do you think?

Reply:
In Codespaces, we can achieve the same by requiring python as a feature (with specific version) instead of installing via system/apt. Please see the changes in latest commit, if it is OK. One minor downside is that it builds python from the source, so it is slightly slower than installing pre-built binaries via apt.

Reply:
Sounds good